inst/examples/build-a-box/R/mod-global-controls.R

```diff
@@ -0,0 +1,116 @@
+ui_global_controls <- function(id) {
+  ns <- shiny::NS(id)
+
+  tagList(
+    layout_columns(
+      class = "align-items-end",
+      selectizeInput(
+        ns("theme_style"),
+        "Theme style",
+        list(
+          "All" = "all",
+          "Semantic Colors" = list(
+            "Background" = "semantic-bg",
+            "Text" = "semantic-fg"
+          ),
+          "Theme Colors" = list(
+            "Background" = "colors-bg",
+            "Text" = "colors-fg"
+          ),
+          "Vibrant" = list(
+            "Gradient" = "gradient"
+          )
+        )
+      ),
+      div(
+        class = "btn-group",
+        role = "group",
+        "aria-label" = "Theme actions",
+        shuffleButton(ns("random_theme"), "Theme"),
+        shuffleButton(ns("random_stat"), "Stats"),
+        actionButton(ns("reset_theme"), "Reset")
+      )
+    ),
+    layout_columns(
+      class = "align-items-start",
+      div(
+        selectInput(
+          ns("showcase_item"),
+          "Showcase Item",
+          c("", "plot", "icon"),
+          "plot"
+        ),
+        conditionalPanel(
+          "input.showcase_item == 'icon'",
+          ns = ns,
+          shuffleButton(ns("random_icon"), "Icons")
+        )
+      ),
+      selectInput(
+        ns("showcase_layout"),
+        "Showcase Layout",
+        c("", "left center", "top right", "bottom"),
+        "left center"
+      )
+    )
+  )
+}
+
+server_global_controls <- function(input, output, sessions, one, two, three) {
+  observeEvent(input$random_theme, {
```

Comment:
Currently, if you choose a new theme style, the value boxes don't update, and it's not necessarily obvious you need to click random theme to apply the new style

```suggestion
  observeEvent({input$random_theme, input$theme_style}, {
```

Reply:
Thanks I noticed this too. It needs to be `c(input$random_theme, input$theme_style)` though and also requires `ignoreInit = TRUE`